engine/orders.go

```diff
@@ -17,47 +17,120 @@ import (
 var (
 	OrderManagerDelay      = time.Second * 10
 	ErrOrdersAlreadyExists = errors.New("order already exists")
+	ErrOrderNotFound       = errors.New("order does not exist")
 )
 
-func (o *orderStore) Get() map[string][]order.Detail {
+// get returns all orders for all exchanges
+// should not be exported as it can have large impact if used improperly
+func (o *orderStore) get() map[string][]*order.Detail {
 	o.m.Lock()
 	defer o.m.Unlock()
 	return o.Orders
 }
 
+// GetByExchangeAndID returns a specific order by exchange and id
+func (o *orderStore) GetByExchangeAndID(exchange, id string) (*order.Detail, error) {
+	o.m.Lock()
+	defer o.m.Unlock()
+	r, ok := o.Orders[exchange]
+	if !ok {
+		return nil, ErrExchangeNotFound
+	}
+
+	for x := range r {
+		if r[x].ID == id {
+			return r[x], nil
+		}
+	}
+	return nil, ErrOrderNotFound
+}
+
+// GetByExchange returns orders by exchange
+func (o *orderStore) GetByExchange(exchange string) ([]*order.Detail, error) {
+	o.m.Lock()
```

Comment:
Is it worth having these are rwmutex that way multiple goroutines can get read locks?
nothing is updated within GetByExchange() so it should be safe for multiple reads

Reply:
That's a good point. I'll change it